pkg/alertmanager/alertmanager.go

```diff
@@ -606,7 +607,16 @@ func (am *Alertmanager) buildIntegrationsMap(gCfg *config.GlobalConfig, nc []*de
 				integration: integrationName,
 			}
 
-			return newRateLimitedNotifier(notifier, rl, 10*time.Second, am.rateLimitedNotifications.WithLabelValues(integrationName))
+			notifier = newRateLimitedNotifier(notifier, rl, 10*time.Second, am.rateLimitedNotifications.WithLabelValues(integrationName))
+		}
+		if am.cfg.EnableNotifyHooks {
```

Comment:
Going to swap these round so that rate limits are applied before hooks are invoked.